colour/graph/conversion.py

```diff
@@ -0,0 +1,803 @@
+    """
+    if is_networkx_installed(raise_exception=True):  # pragma: no cover
+        path = nx.shortest_path(CONVERSION_GRAPH, source, target)
+
+        return [
+            CONVERSION_GRAPH.get_edge_data(a, b)['conversion_function']
+            for a, b in zip(path[:-1], path[1:])
+        ]
+
+
+@domain_range_scale('1')
+def convert(a, source, target, verbose_parameters=None, **kwargs):
+    """
+    Converts given object :math:`a` from source colour representation to target
+    colour representation using the automatic colour conversion graph.
+
+    The conversion is performed by finding the shortest path in a
+    `NetworkX <https://networkx.github.io/>`_ :class:`DiGraph` class instance.
+
+    The conversion path adopts the **'1'** domain-range scale and the object
+    :math:`a` is expected to be *soft* normalised accordingly. For example,
+    *CIE XYZ* tristimulus values arguments for use with the *CAM16* colour
+    appearance model should be in domain `[0, 1]` instead of the domain
+    `[0, 100]` used with the **'Reference'** domain-range scale. The arguments
+    are typically converted as follows:
+
+    -   *Scalars* in domain-range `[0, 10]`, e.g *Munsell Value* are
+        scaled by *10*.
+    -   *Percentages* in domain-range `[0, 100]` are scaled by *100*.
+    -   *Degrees* in domain-range `[0, 360]` are scaled by *360*.
+    -   *Integers* in domain-range `[0, 2**n -1]` where `n` is the bit
+        depth are scaled by *2**n -1*.
+
+    See the `Domain-Range Scales <../basics.html#domain-range-scales>`_ page
+    for more information.
+
+    Parameters
+    ----------
+    a : array_like or numeric or SpectralDistribution
+        Object :math:`a` to convert.
+    source : unicode
+        Source colour representation, i.e. the source node in the automatic
+        colour conversion graph.
+    target : unicode
+        Target colour representation, i.e. the target node in the automatic
+        colour conversion graph.
+    verbose_parameters : dict, optional
+        Verbose parameters to describe the conversion.
+
+    Other Parameters
+    ----------------
+    \\**kwargs : dict, optional
+        Please refer to the documentation of the supported definitions.
+        Arguments for the definitions are passed as keyword arguments with
+        values set as dictionaries. For example, in the conversion from
+        spectral distribution to *sRGB* colourspace, passing arguments to the
+        `colour.sd_to_XYZ` definitions is done as follows::
+
+            convert(sd, 'Spectral Distribution', 'sRGB', sd_to_XYZ={\
+'illuminant': illuminant})
+
+    Returns
+    -------
+    ndarray or numeric or SpectralDistribution
+        Converted object :math:`a`.
+
+    Notes
+    -----
+    -   Various defaults have been systematically adopted compared to the
+        low-level *Colour* API:
+
+        -   The domain-range scale is **'1'**.
+        -   The default illuminant is *CIE Standard Illuminant D Series* *D65*.
```

Comment:
Is this also the expected/required input illuminant? If so this could merit additional highlighting.

Reply:
It is not required but more the default wherever the underlying definitions require an illuminant. That said it is effectively a good thing to state that the data passed to `colour.convert` is expected to be under **D65**.